<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>613</width>
    <height>416</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QtJobs</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QGroupBox { font-weight: bold; }</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_7">
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>File list</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="leftMargin">
         <number>1</number>
        </property>
        <property name="topMargin">
         <number>4</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>6</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_10">
            <property name="font">
             <font>
              <italic>true</italic>
             </font>
            </property>
            <property name="text">
             <string>Please add files to be processed by the command that you specify on the next tab.</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QListWidget" name="fileList">
          <property name="selectionMode">
           <enum>QAbstractItemView::ExtendedSelection</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QPushButton" name="pushButton">
            <property name="text">
             <string>Add files</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_2">
            <property name="text">
             <string>Remove selected</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_3">
            <property name="text">
             <string>Remove all</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="title">
           <string>Profile settings</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="0" column="0">
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Profile:</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QComboBox" name="comboProfile">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QPushButton" name="pushButton_6">
               <property name="text">
                <string>Load</string>
               </property>
              </widget>
             </item>
             <item row="0" column="3">
              <widget class="QPushButton" name="pushButton_8">
               <property name="text">
                <string>Remove</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Save as:</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="profileEdit">
               <property name="toolTip">
                <string>If you leave this field empty, the selected profile will be overwritten.</string>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QPushButton" name="pushButton_7">
               <property name="text">
                <string>Save</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>Output file name settings</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <property name="margin">
            <number>2</number>
           </property>
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="1" colspan="2">
              <widget class="QLineEdit" name="extEdit">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="toolTip">
                <string>Enter the extension without the &quot;.&quot; prefix</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0" colspan="3">
              <widget class="QRadioButton" name="radioSame">
               <property name="text">
                <string>Save each file to the original file's directory</string>
               </property>
               <attribute name="buttonGroup">
                <string>buttonGroup</string>
               </attribute>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QCheckBox" name="extCb">
               <property name="text">
                <string>Change the file extension to:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QRadioButton" name="radioOther">
               <property name="text">
                <string>Save all files to the following directory:</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
               <attribute name="buttonGroup">
                <string>buttonGroup</string>
               </attribute>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLineEdit" name="outDirEdit"/>
             </item>
             <item row="2" column="2">
              <widget class="QPushButton" name="dirBtn">
               <property name="text">
                <string>...</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Command settings</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_9">
           <property name="margin">
            <number>2</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Command:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="commandEdit"/>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'DejaVu Sans'; font-size:10pt; font-weight:600;&quot;&gt;Note&lt;/span&gt;&lt;span style=&quot; font-family:'DejaVu Sans'; font-size:10pt;&quot;&gt;: Variable &lt;/span&gt;&lt;span style=&quot; font-family:'DejaVu Sans'; font-size:10pt; font-style:italic;&quot;&gt;$FILE&lt;/span&gt;&lt;span style=&quot; font-family:'DejaVu Sans'; font-size:10pt;&quot;&gt; will be substituted for the input file name and &lt;/span&gt;&lt;span style=&quot; font-family:'DejaVu Sans'; font-size:10pt; font-style:italic;&quot;&gt;$OFILE&lt;/span&gt;&lt;span style=&quot; font-family:'DejaVu Sans'; font-size:10pt;&quot;&gt; for the output one.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>Launch settings</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="margin">
            <number>2</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QCheckBox" name="threadcountCb">
               <property name="text">
                <string>Use this thread count:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="threadBox">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>64</number>
               </property>
               <property name="value">
                <number>2</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_5">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>(instead of using the auto-detection)</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="runBtn">
             <property name="styleSheet">
              <string notr="true">font-weight: bold;</string>
             </property>
             <property name="text">
              <string>RUN!</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>1</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Progress</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="topMargin">
         <number>8</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Total progress:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="progressBar">
            <property name="maximum">
             <number>1</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="format">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="stopBtn">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>This will let finish the running commands and prevent the rest of them from starting</string>
            </property>
            <property name="text">
             <string>Stop</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="killBtn">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="toolTip">
             <string>This will immediately kill all the running commands and prevent the rest of them from starting</string>
            </property>
            <property name="text">
             <string>KILL!</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>List of running commands:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="commandList"/>
        </item>
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>List of successfully processed files:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="processedList"/>
        </item>
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>List of commands that returned a non-zero exit code:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="errorList"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>radioOther</sender>
   <signal>toggled(bool)</signal>
   <receiver>outDirEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>215</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>581</x>
     <y>230</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioOther</sender>
   <signal>toggled(bool)</signal>
   <receiver>dirBtn</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>159</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>623</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioSame</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>forceExtensionChange(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>577</x>
     <y>203</y>
    </hint>
    <hint type="destinationlabel">
     <x>616</x>
     <y>222</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>extCb</sender>
   <signal>toggled(bool)</signal>
   <receiver>extEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>623</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addFiles()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>56</x>
     <y>467</y>
    </hint>
    <hint type="destinationlabel">
     <x>547</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeSelectedFiles()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>301</x>
     <y>460</y>
    </hint>
    <hint type="destinationlabel">
     <x>397</x>
     <y>23</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>fileList</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>432</x>
     <y>462</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>121</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_8</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeSelectedProfile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>590</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>638</x>
     <y>45</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_7</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveProfile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>502</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>638</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadSelectedProfile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>510</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>638</x>
     <y>34</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>runJobs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>357</x>
     <y>458</y>
    </hint>
    <hint type="destinationlabel">
     <x>394</x>
     <y>501</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>stopJobs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>515</x>
     <y>31</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>-2</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>killBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>killJobs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>580</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>577</x>
     <y>-5</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addFiles()</slot>
  <slot>removeSelectedFiles()</slot>
  <slot>forceExtensionChange(bool)</slot>
  <slot>removeSelectedProfile()</slot>
  <slot>saveProfile()</slot>
  <slot>loadSelectedProfile()</slot>
  <slot>runJobs()</slot>
  <slot>stopJobs()</slot>
  <slot>killJobs()</slot>
 </slots>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
